File: Problemset/reverse-integer/reverse-integer.js

// @Title: 整数反转 (Reverse Integer)
// @Author: 846188037@qq.com
// @Date: 2020-06-08 21:01:23
// @Runtime: 92 ms
// @Memory: 34.7 MB

/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(x) {
    x=String(x);
    var s="";
    if(x.charAt(0)=="-")
    {  
        for(var i=x.length-1;i>0;i--)
        {
            s+=x.charAt(i)
        }
        s="-"+s;

    }
    else{
        for(var i=x.length-1;i>-1;i--)
        {
            s+=x.charAt(i)
        }

    }
    s=Number(s);
    if(s<Math.pow(-2,31)||s>=Math.pow(2,31))
    {
        s=0;
    }

   
    return s

};

File: Problemset/reverse-integer/reverse-integer.go

// @Title: 整数反转 (Reverse Integer)
// @Author: 846188037@qq.com
// @Date: 2021-03-18 10:35:25
// @Runtime: 0 ms
// @Memory: 2.1 MB

func reverse(x int) int {
    flag:=x<0
    if flag{
        x=-x
    }
    res:=0
    for x>0{
        res=res*10+(x%10)
        x=x/10
        if res < -2147483648 || res > 2147483648{
            return 0
        }
    }
    if flag{
        return -res
    }
    return res

    
    
}
